File: src/main/resources/html/app/image.js
/* - image.js */

function Image() {

	var self			= this;

	self.width			= null;		//integer
	self.height			= null;		//integer
	self.url			= null;		//string

	return self;
}

Image.parseArray		= function(images) {
	if (images) {
		var parsedImages	= [];
		for (var i=0; i < images.length; ++i) {
			parsedImages.push(Image.parse(images[i]));
		}
		return parsedImages;
	}

	return null;
};


Image.parse				= function(rawJson) {
	if (rawJson) {
		var image		= new Image();
		image.width		= rawJson.width;
		image.height	= rawJson.height;
		image.url		= rawJson.url;
		return image;
	}

	return null;
};

File: src/main/java/com/joey/jseach/core/Image.java
package com.joey.jseach.core;

import com.google.gson.JsonObject;
import com.joey.jseach.network.JsonSerializable;

public class Image implements JsonSerializable {
	private final int width;
	private final int height;
	private final String url;


	public Image(int width, int height, String url) {
		this.width = width;
		this.height = height;
		this.url = url;
	}

	@Override
	public String toString() {
		return String.format("(width:%d, height:%d, url:%s)", width, height, url);
	}

	@Override
	public JsonObject toJson() {
		JsonObject json = new JsonObject();
		json.addProperty("width", width);
		json.addProperty("height", height);
		json.addProperty("url", url);
		return json;
	}
}
